sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -1920,6 +1920,208 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
             }
         }
         #endregion ScheduleDeletion
+
+        #region Query
+        /// <summary>
+        /// The <see cref="Query"/> API returns the
+        /// result of a query against the blob.
+        /// </summary>
+        /// <param name="query">
+        /// The query.
+        /// </param>
+        /// <param name="inputTextConfiguration">
+        /// Optional input text configuration.
+        /// </param>
+        /// <param name="outputTextConfiguration">
+        /// Optional output text configuration.
+        /// </param>
+        /// <param name="errorReceiver">
+        /// Optional error receiver.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional request conditions.
+        /// </param>
+        /// <param name="progressReceiver">
+        /// Optional progress receiver.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        /// <returns>
+        /// A <see cref="Response{BlobQueryInfo}"/>.
+        /// </returns>
+        public virtual Response<BlobDownloadInfo> Query(
```

Comment:
Probably want to take Ted's API view comment, and switch to an optional parameters group.

Reply:
+1